client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -99,6 +101,39 @@ var CheckoutThankYou = React.createClass( {
 		return null;
 	},
 
+	goBackUrl() {
+		var purchases;
+		if ( this.isDataLoaded() ) {
+			purchases = getPurchases( this.props );
+
+			function shouldGoBackToPlans() {
+				return purchases.some( isJetpackPremium ) || purchases.some( isJetpackBusiness ) || purchases.some( isPremium ) || purchases.some( isBusiness );
+			}
+
+			function shouldGoBackToDomainManagment() {
+				return purchases.some( isDomainRegistration ) || purchases.some( isDomainMapping ) || purchases.some( isSiteRedirect ) || purchases.some( isDomainRedemption );
+			}
+
+			function shouldGoBackToDomainManagmentEmail() {
+				return purchases.some( isGoogleApps );
+			}
+
+			if ( shouldGoBackToPlans( purchases ) ) {
+				return '/plans/' + this.props.selectedSite.slug;
+			} else if ( shouldGoBackToDomainManagment( purchases ) ) {
+				return '/domains/manage/' + this.props.selectedSite.slug;
+			} else if ( shouldGoBackToDomainManagmentEmail( purchases ) ) {
+				return '/domains/manage/email/' + this.props.selectedSite.slug;
+			}
+		}
+
+		return '/stats/' + this.props.selectedSite.slug;
+	},
+
+	goBack: function() {
+		page( this.goBackUrl() );
```

Comment:
What is the rationale behind making these two functions? I think it'd be clearer if all of the logic lived in `goBack`.